src/latexdocvisitor.cpp

```diff
@@ -1307,7 +1311,11 @@ void LatexDocVisitor::visitPost(DocHtmlCell *c)
     setInColSpan(FALSE);
     m_t << "}";
   }
-  if (!c->isLast()) m_t << "&";
+  if (!c->isLast()) {
+    m_t << "&";
+    if (!c->isHeading()) m_t << "\\hspace{0pt}";
+  }
+
```

Comment:
When not mistaken I saw this proposed patch also as part of #8025. Probably it is better to have as a separate proposed patch, but in that case it should be removed from #8025